specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2024-07-02-preview/managedClusters.json

```diff
@@ -7717,6 +7717,28 @@
           "type": "boolean",
           "title": "Whether to auto install GPU drivers or not.",
           "description": "The default value is true when the vmSize of the agent pool contains a GPU, false otherwise. GPU Driver Installation can only be set true when VM has an associated GPU resource. Setting this field to false prevents automatic GPU driver installation. In that case, in order for the GPU to be usable, the user must perform GPU driver installation themselves."
+        },
+        "driverType": {
+          "type": "string",
+          "enum": [
+            "GRID",
+            "CUDA"
+          ],
+          "x-ms-enum": {
+            "name": "driverType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "GRID",
+                "description": "Install the GRID driver for the GPU, suitable for applications requiring virtualization support."
+              },
+              {
+                "value": "CUDA",
+                "description": "Install the CUDA driver for the GPU, optimized for computational tasks in scientific computing and data-intensive applications."
+              }
+            ]
+          },
+          "description": "Specify the type of GPU driver to install when creating Windows agent pools. If not provided, or if you are creating Linux agent pools, AKS selects the driver based on system compatibility. This cannot be changed once the AgentPool has been created."
```

Comment:
```suggestion
          "description": "Specify the type of GPU driver to install when creating Windows agent pools. If not provided, AKS selects the driver based on system compatibility. This cannot be changed once the AgentPool has been created."
```

Reply:
The driver selection feature is not available to Linux. Even if a driver type is provided when creating a linux gpu ap, we return GPUDriverTypeSelectionNotAllowedForLinux error.

Reply:
I think the word "Windows" also needs to be removed because Linux is also sharing this property.

Reply:
Actually I remember now, this property only works on Windows pools for now:

I think this should say:
> Specify the type of GPU driver to install when creating Windows agent pools. If not provided, AKS selects the driver based on system compatibility. This cannot be changed once the AgentPool has been created. This cannot be set on Linux AgentPools. For Linux AgentPools, the driver is selected based on system compatibility.